src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1209,6 +1209,8 @@ public virtual bool TryValidateModel([NotNull] object model, string prefix)
                 throw new InvalidOperationException(message);
             }
 
+            ModelStateDictionaryUtility.ClearModelStateDictionary(ModelState, model.GetType());
```

Comment:
Also I thought the decision was to invalidate the tree for the model type _ONLY_ and not the entire state. The name of the method tells me that it will clear everything.